tests/sync/app.cpp

```diff
@@ -733,6 +738,75 @@ TEST_CASE("app: UserAPIKeyProviderClient integration", "[sync][app]") {
 
 }
 
+TEST_CASE("app: refresh access token integration tests", "[sync][app]") {
+
+    SECTION("refresh custom data integration expect success") {
+        
+        auto email = util::format("%1@%2.com", random_string(15), random_string(15));
+        auto password = util::format("%1", random_string(15));
+
+        std::unique_ptr<GenericNetworkTransport> (*factory)() = []{
+            return std::unique_ptr<GenericNetworkTransport>(new IntTestTransport);
+        };
+    
+        auto app = App(App::Config{"translate-utwuv", factory});
+
+        static const std::string base_path = realm::tmp_dir();
+        auto tsm = TestSyncManager(base_path);
+        bool processed = false;
+
+        app.provider_client<App::UsernamePasswordProviderClient>().register_email(email,
+                                                                                  password,
+                                                                                  [&] (Optional<AppError> error) {
+            CHECK(!error);
+        });
+
+        app.log_in_with_credentials(AppCredentials::username_password(email, password),
+                               [&](std::shared_ptr<SyncUser> user, Optional<app::AppError> error) {
+            CHECK(user);
+            CHECK(!error);
+        });
+        
+        auto previous_access_token = SyncManager::shared().get_current_user()->access_token();
+        
+        // If we do this call within the same second, the response for the access token will be cached
```

Comment:
Who controls that?

Reply:
Its a server side issue, because we are testing the call within the same second. The way around it is to sleep at least a second